examples/video_upload_multipart.py

```diff
@@ -0,0 +1,127 @@
+
+
+class MultipartUploader(object):
+    # set default buffer size to 10 MB
+    BUFFER_SIZE = 1000000
+
+    def __init__(self, video_file_path):
+        self._file_path = video_file_path
+        self._file_size = os.stat(video_file_path).st_size
+        self._file_name = os.path.split(video_file_path)[1]
+        self._setup()
+
+    def _setup(self):
+        """
+        Initializes the MultipartUploader object for use.  Namely, this method configures the upload
+        URL and all relevant parameters and headers for the POST requests to be made.
+
+        :param filename: <str> the name of the video file
+        """
+        try:
+            self._jwplatform_client = Client(JW_API_KEY, JW_API_SECRET)
+            self._response = self._jwplatform_client.videos.create(upload_method='multipart', title=self._file_name)
+            self._filename = self._file_name
+
+            self._upload_url = '{}://{}{}'.format(
+                self._response['link']['protocol'],
+                self._response['link']['address'],
+                self._response['link']['path']
+            )
+            logging.info('Upload URL to be used: {}'.format(self._upload_url))
+            self._query_parameters = self._response['link']['query']
+            self._query_parameters['api_format'] = 'json'
+            self._headers = {'X-Session-ID': self._response['session_id']}
+            self._offset = 0
+
+        except JWPlatformError as e:
+            logging.exception('An error occurred during the uploader setup. Check that your API keys are properly '
+                              'set up in your environment, and ensure that the video file path exists.')
+            raise
+
+    def _upload_chunk(self, chunk):
+        """
+        Handles the POST request needed to upload a single portion of the video file.
+        Serves as a helper method for upload().
+
+        The offset used to determine where a chunk begins and ends is updated in the course of
+        this method's execution.
+
+        :param chunk: <byte[]> the raw bytes of data from the video file
+        """
+
+        begin_chunk = self._offset
+        # the last byte in the range is INCLUSIVE, hence the -1
+        end_chunk = begin_chunk + len(chunk) - 1
+
+        logging.info("begin_chunk / end_chunk = {} / {}".format(begin_chunk, end_chunk))
+        self._headers.update({'X-Content-Range': 'bytes {}-{}/{}'.format(begin_chunk, end_chunk, self._file_size)})
+        self._headers.update({'Content-Disposition': 'attachment; filename="{}"'.format(self._filename)})
+        self._headers.update({'Content-Type': 'application/octet-stream'})
+        self._headers.update({'Content-Length': str((end_chunk - begin_chunk) + 1)})
+
+        try:
+            response = requests.post(self._upload_url, params=self._query_parameters, headers=self._headers, data=chunk)
+            response.raise_for_status()
+        except requests.exceptions.RequestException:
+            raise
```

Comment:
The caller catches the exception and logs a message for it.  I was having trouble with logging something here because I got the exception message to print out twice; I removed the logging to eliminate the duplication.  Although I suppose if I didn't use `logging.exception` then it would be OK.